stream.go

```diff
@@ -1481,6 +1481,8 @@ type serverStream struct {
 	comp   encoding.Compressor
 	decomp encoding.Compressor
 
+	sendCompressorName string
```

Comment:
Why do we need to store this field? Can't we get it from `comp.Type()`?

Reply:
We can't compare it with `comp.Type()` as the stream could either have `cp Compressor` or `comp encoding.Compressor` set.